auditlog/audit_log_blackbox_test.go

```diff
@@ -0,0 +1,205 @@
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/errors"
+
+	"github.com/fabric8-services/admin-console/auditlog"
+	"github.com/fabric8-services/admin-console/configuration"
+	"github.com/fabric8-services/fabric8-common/resource"
+	testsuite "github.com/fabric8-services/fabric8-common/test/suite"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RepositoryBlackBoxTest struct {
+	testsuite.DBTestSuite
+	repo auditlog.Repository
+}
+
+func TestRecordRepository(t *testing.T) {
+	resource.Require(t, resource.Database)
+	config, err := configuration.New()
+	require.NoError(t, err)
+	suite.Run(t, &RepositoryBlackBoxTest{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *RepositoryBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = auditlog.NewRepository(s.DB)
+}
+
+func (s *RepositoryBlackBoxTest) TestCreateRecord() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		before := time.Now()
+		auditLog := auditlog.AuditLog{
+			EventTypeID: auditlog.UserSearch,
+			IdentityID:  uuid.NewV4(),
+			EventParams: auditlog.EventParams{},
+		}
+		// when
+		err := s.repo.Create(context.Background(), &auditLog)
+		// then
+		require.NoError(t, err)
+		assert.NotEqual(t, uuid.NullUUID{}, auditLog.ID)
+		assert.True(t, auditLog.CreatedAt.After(before)) // "is after before". hahahaha....
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+
+		t.Run("missing event type", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				IdentityID:  uuid.NewV4(),
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
+		})
+
+		t.Run("missing identity id", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				EventTypeID: auditlog.UserSearch,
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
+		})
+	})
+}
+
+func (s *RepositoryBlackBoxTest) TestLoadByID() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		auditLog := auditlog.AuditLog{
+			EventTypeID: auditlog.UserSearch,
+			IdentityID:  uuid.NewV4(),
+			EventParams: auditlog.EventParams{
+				"idx": 1,
+			},
+		}
+		err := s.repo.Create(context.Background(), &auditLog)
+		require.NoError(t, err)
+		// when
+		result, err := s.repo.LoadByID(context.Background(), auditLog.ID)
+		// then
+		require.NoError(t, err)
+		// comparing 'CreatedAt' may cause troubles b/c of nanosecond roundings, so let's just verify that the result ID is the one expected
+		assert.Equal(t, auditLog.ID, result.ID)
+
+	})
+	s.T().Run("not found", func(t *testing.T) {
+		// when
+		_, err := s.repo.LoadByID(context.Background(), uuid.NewV4())
+		// then
+		require.Error(t, err)
+		assert.IsType(t, errors.NotFoundError{}, err)
```

Comment:
We could move https://github.com/fabric8-services/fabric8-auth/blob/49922e5bfbc19129d6e193f6e52050fd18230633/test/error.go#L11 to common
and instead of three asserts/require (Error, type, message) we could have one assert here - https://github.com/fabric8-services/fabric8-auth/blob/9d855d80053b54776839a5316c6c609500dc072d/authorization/resourcetype/repository/resource_type_scope_blackbox_test.go#L65

Reply:
yes, indeed. I've opened https://github.com/fabric8-services/fabric8-common/issues/35 for this matter